src/vellum/workflows/monitoring/decorators.py

```diff
@@ -0,0 +1,214 @@
+"""Monitoring decorators for workflow tracing."""
```

Comment:
[nit] `src/vellum/workflows/events` is basically our monitoring module, would be nice to colocate there